lightningd/plugin.c

```diff
@@ -1254,6 +1254,7 @@ const char *plugin_send_getmanifest(struct plugin *p)
 	p->stdin_conn = io_new_conn(p, stdin, plugin_stdin_conn_init, p);
 	req = jsonrpc_request_start(p, "getmanifest", p->log,
 				    plugin_manifest_cb, p);
+	json_add_bool(req->stream, "allow-deprecated-apis", deprecated_apis);
```

Comment:
I'm almost sure no plugin library will crash if there is an additional entry in a JSON object.

Reply:
Our docs describe it as being called without any params, so a third-party plugin could. theoretically, check that params is either an empty object or an empty array, and fail if it gets a non-empty one.

At the very least, we should update our docs to say that "`getmanifest` may provide additional parameters; you should ignore any parameters you do not understand."  We should probably also indicate that the `params` of `getmanifest` will always be a JSON object, as JSON-RPC 2.0 allows either an object or an array.

Reply:
Funny that you mention it: `pyln`s implementation of the plugin interface can fail if the hook, notification, or method gets called with an unknown argument, unless they add `**kwargs` to their signature, because pyln is "clever" and unpacks params from the JSON-RPC call and matches them with the function signature (failing if it can't match them all).

Reply:
Indeed.  Fortunately `pyln` intercepts the `getmanifest` call and uses `**kwargs`, so at least typical use of `pyln` should work (though if the python plugin gets clever and decides to derive from the `pyln` classes and override `getmanifest`...).

A fallback we could do is if the `getmanifest` call fails when we provide *any* argument we could retry the same `getmanifest` without arguments, but only if `deprecated_apis` is `true`.  Then put that behavior in a deprecation cycle, and update our docs to make it very clear that `getmanifest` can give additional params in the future and plugins should specifically ignore params they do not understand.

Reply:
Yes, note that one of our primitive test plugins crashed due to this!

I like @ZmnSCPxj 's idea of putting it in the deprecation cycle, since it's default-true anyway.  Will change.